### PR TITLE
Print error log when tests fail

### PR DESCRIPTION
**Pull request by oflatt**
Currently, prints 50 lines of the error log each time a test fails.

### CODE REVIEW

File: tests/test_functions.py [hunk 1]
```diff
@@ -223,3 +226,31 @@ def cleanup_check():
     Checks to make sure all endpoints were tested."""
 
     test_state.cleanup_check()
+
+
+def run_bash(command):
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+def file_tail(filename, length):
```
Comment (zachzundel) [inline tests/test_functions.py]: Why not read the whole file into memory and then print the lines? failure in os.popen can be hard to handle

Comment (oflatt) [reply tests/test_functions.py]: Good idea- I was probably over-optimizing

File: tests/run_sboltestrunner.sh [hunk 2]
```diff
@@ -38,5 +38,7 @@ java -jar SBOLTestRunner/target/SBOLTestRunner-0.0.1-SNAPSHOT-withDependencies.j
 
 exitcode=$?
 if [ $exitcode -ne 0 ]; then
```
Comment (zachzundel) [inline tests/run_sboltestrunner.sh]: Does SBOLTestRunner exit with an error code if it fails I’m not sure it does.

File: tests/fetch_logs.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+
+# if length is negative, returns all lines
+def file_tail(filename, length):
+    with open(filename, "r") as file_to_read:
+        lines = file_to_read.readlines()
+        if length < len(lines) and length >= 0:
+            return "".join(lines[len(lines)-length:])
```
Comment (zachzundel) [inline tests/fetch_logs.py]: Does this print new lines or does it squish everything onto the same line?

File: tests/run_sboltestrunner.sh (hunk 2 above)
Comment (oflatt) [reply tests/run_sboltestrunner.sh]: I have tested it and looked at the code, I believe it does.

**Comment by oflatt** [comment]
A new problem is that I need to parse the date on the error logs to find the newest one to print.

File: tests/fetch_logs.py (hunk 3 above)
Comment (oflatt) [reply tests/fetch_logs.py]: This prints new lines.

**Comment by zachzundel** [comment]
Would it be easier to use `docker logs testsuiteproject_synbiohub_1` ?

**Comment by oflatt** [comment]
Probably, but it's nice to have a local version of the logs. I've implemented the search for the newest log.

**Pull request merged**